### PR TITLE
Updated with cleaning of expectation counts that are zero

### PR DESCRIPTION
**Pull request by glenoliff**
When an engineer makes the mistake of adding an expectation of a zero count when using `only_load_models`, they get an odd error message with an empty difference section. This PR is to simply trim off the 0 count entries before the comparison takes place between actual and expected hashes.

### CODE REVIEW

File: lib/ar_query_matchers.rb [hunk 1]
```diff
@@ -7,6 +7,20 @@
...
   module ArQueryMatchers
+    class Utility
+      def self.clean_expected(expected)
+        expected_cleaned = expected.select { |k,v| v > 0 }
+
+        # On the fence about whether or not to include a warning message. Folks writing tests should be aware that they
+        #   should remove the 0 count expectations to keep the tests clean. Problem is it makes the output for this test
+        #   suite noisy. I could go either way on this
+        # amount_cleaned = expected.size - expected_cleaned.size
+        # puts "Removed #{amount_cleaned} 0 count expectation(s), please remove from your test to hide this message" if amount_cleaned > 0
```
Comment (glenoliff) [inline lib/ar_query_matchers.rb:18]: I like the idea of informing the engineer that their test has a superfluous zero count. However it makes the tests local to this gem noisy. Happy to go either way on this one. Obviously, I'd clean up this method if we choose to not notify, as this method will go down to a one-liner. If we do choose to notify, I couldn't use pluralize, given we're not in Rails land, and would appreciate a way to do the pluralization without rolling our own (not hard but duplicative).

Comment (mzruya) [reply lib/ar_query_matchers.rb:18]: I can also see an argument for why you'd like to assert 0:
```ruby
expect { do_something }.to load_models('Employee' => 0, 'Company => 1)
```
vs

```ruby
expect { do_something }.to load_models('Company => 1)
```

The former allows you to convey the intent that you are explicitly expecting 0 employee queries.
In the latter it's implicit.

File: lib/ar_query_matchers.rb [hunk 2]
```diff
@@ -7,6 +7,20 @@
 
 module ArQueryMatchers
   module ArQueryMatchers
+    class Utility
+      def self.clean_expected(expected)
```
Comment (mzruya) [inline lib/ar_query_matchers.rb:11]: perhaps a more concrete name:
```suggestion
      def self.remove_superfluous_expectations(expected)
```

File: spec/ar_query_matchers/query_matchers_spec.rb [hunk 3]
```diff
@@ -90,6 +90,14 @@ def updates(amount)
       end.to only_update_models('MockUser' => 5)
     end
 
+    it 'succeeds after cleaning the 0 count expectation' do
```
Comment (mzruya) [inline spec/ar_query_matchers/query_matchers_spec.rb:93]: awesome!

**Comment by mzruya** [comment]
@glenoliff, can you also update the CHANGELOG file?

**Comment by glenoliff** [comment]
> @glenoliff, can you also update the CHANGELOG file?

Done

File: lib/ar_query_matchers.rb (hunk 2 above)
Comment (glenoliff) [reply lib/ar_query_matchers.rb]: Cool, done

File: VERSION [hunk 4]
```diff
@@ -1 +1 @@
-0.5.0
\ No newline at end of file
+0.5.1
```
Comment (glenoliff) [inline VERSION:1]: Since it's a minor change, figured this was the correct path for the next version number. Happy to change to whatever.

**Review by mzruya — APPROVED** [review]

**Pull request merged**